src/OrchardCore.Modules/OrchardCore.Layers/Services/LayerFilter.cs

```diff
@@ -146,5 +152,19 @@ public async Task OnResultExecutionAsync(ResultExecutingContext context, ResultE
 
             await next.Invoke();
         }
+
+        private async Task<IEnumerable<LayerMetadata>> FilterWidgetsByCultureAsync(IEnumerable<LayerMetadata> widgets)
+        {
+            var orchardHelper = _orchardHelper;
+            var widgetsWithCulture = await Task.WhenAll(widgets.Select(async w =>
+            {
+                var culture = await orchardHelper.GetContentCultureAsync(w.ContentItem);
+                return new { Widget = w, Culture = culture };
```

Comment:
I don't think this anonymous object is necessary. You can use IAsyncEnumerable instead by iterating over each widget.